### PR TITLE
Updated Travis jobs to include OSD and ROSA builds

### PR DESCRIPTION
**Pull request by gaurav-nelson**
Updtaes for the Travis CI build scripts:
- Kicks in 3 stages = 3 different VMs running in parallel
- Checks OCP, OSD, and ROSA builds
- If the PR author is `openshift-cherrypick-robot` runs the Automerge script otherwise skips it.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,19 +1,41 @@
 language: python
-
 cache: pip
+env:
+  - PR_AUTHOR=${TRAVIS_PULL_REQUEST_SLUG::-15}
```
Comment (gaurav-nelson) [inline .travis.yml:4]: Allows us to skip the automerge build if the PR_AUTHOR is not `openshift-cherrypick-robot`. Check line 36
- fails silently https://app.travis-ci.com/github/openshift/openshift-docs/builds/245020194 VS https://app.travis-ci.com/github/openshift/openshift-docs/builds/245019610

**Comment by gaurav-nelson** [comment]
@pneedle-rh PTAL 

Feel free to use this PR or modify your original PR's.

File: .travis.yml [hunk 2]
```diff
@@ -1,19 +1,41 @@
...
+    - # stage name not required, will continue to use `build`
+      before_install:
+        - gem install asciidoctor
+        - gem install asciidoctor-diagram
+      install:
+        - pip3 install pyyaml
+        - pip3 install aura.tar.gz
+      script:
+        - python3 build.py --distro openshift-rosa --product "Red Hat OpenShift Service on AWS" --version 4 --no-upstream-fetch && python3 makeBuild.py
+    - stage: automerge
```
Comment (pneedle-rh) [inline .travis.yml:35]: @gaurav-nelson does the automerge stage only trigger if all of the previous stages report a success?

**Comment by pneedle-rh** [comment]
> @pneedle-rh PTAL
> 
> Feel free to use this PR or modify your original PRs.

@gaurav-nelson thank you for developing this! This looks like a much better solution that the one I proposed in https://github.com/openshift/openshift-docs/pull/40591 and https://github.com/openshift/openshift-docs/pull/40592. As such, I have closed those two PRs as obsoleted by this one.

The OSD and ROSA content is added to `main` and cherry picked to `enterprise-4.9` and `enterprise-4.10`. Given that, my assumption is that this PR should be cherry picked to update the .travis.yml` file in those two enterprise branches.

Thanks again!

File: .travis.yml (hunk 2 above)
Comment (gaurav-nelson) [reply .travis.yml:35]: @pneedle-rh Yes. From [Travis Docs](https://docs.travis-ci.com/user/build-stages/):
> In the simplest and most common use case, you can now make one job run only if several other, parallel jobs have completed successfully.

Comment (pneedle-rh) [reply .travis.yml:35]: Excellent! Thank you for confirming that.

**Comment by gaurav-nelson** [comment]
@pneedle-rh Can you add the required labels for this PR ?

**Comment by pneedle-rh** [comment]
@gaurav-nelson in my limited knowledge of Travis, this looks good to me. But, that said, I have not tested the functionality and have just reviewed the overall procedure.

**Comment by pneedle-rh** [comment]
> @pneedle-rh Can you add the required labels for this PR ?

Done.

**Comment by gaurav-nelson** [comment]
@pneedle-rh  
> The OSD and ROSA content is added to `main` and cherry picked to `enterprise-4.9` and `enterprise-4.10`. Given that, my assumption is that this PR should be cherry picked to update the .travis.yml` file in those two enterprise branches.
> 

- I think there should a separate Travis files for 4.9, 4.8 and others which should target building 4.9, 4.8 and so on.

**Comment by pneedle-rh** [comment]
> @pneedle-rh
> 
> > The OSD and ROSA content is added to `main` and cherry picked to `enterprise-4.9` and `enterprise-4.10`. Given that, my assumption is that this PR should be cherry picked to update the .travis.yml` file in those two enterprise branches.
> 
>     * I think there should a separate Travis files for 4.9, 4.8 and others which should target building 4.9, 4.8 and so on.

@gaurav-nelson sounds good. For OSD and ROSA, the checks are only required for `main`, `enterprise-4.9` and `enterprise-4.10`. We did not migrate the OSD and ROSA content to 4.8.

**Comment by ahardin-rh** [comment]
LGTM, but you might want another more trained eye to conduct a separate review. :+1:

**Pull request merged**